pkgs/by-name/qu/qui/package.nix

```diff
@@ -0,0 +1,87 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  stdenvNoCC,
+  nix-update-script,
+  nodejs,
+  pnpm_9,
+  typescript,
+  versionCheckHook,
+}:
+buildGoModule (finalAttrs: {
+  pname = "qui";
+  version = "1.7.0";
+  src = fetchFromGitHub {
+    owner = "autobrr";
+    repo = "qui";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-CbPdngskDCAAhmsj5DPdnviZSWM0bO13Pbe7wRwaNaw=";
+  };
+
+  qui-web = stdenvNoCC.mkDerivation (finalAttrs': {
+    pname = "${finalAttrs.pname}-web";
+    inherit (finalAttrs) src version;
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm_9.configHook
+      typescript
+    ];
+
+    sourceRoot = "${finalAttrs.src.name}/web";
+
+    pnpmDeps = pnpm_9.fetchDeps {
+      inherit (finalAttrs')
+        pname
+        version
+        src
+        sourceRoot
+        ;
+      fetcherVersion = 1;
+      hash = "sha256-94/m0RrGNzxdORidFdsJ9ALUFpQ5CJzQFqHuxhiXoQE=";
+    };
+
+    postBuild = ''
+      pnpm run build
+    '';
+
+    installPhase = ''
+      cp -r dist $out
+    '';
+  });
+
+  vendorHash = "sha256-rmUEFX8UzxEN7XaJ8Zj+kj3z1pwLkq3FTYzbPWnifW0=";
+
+  preBuild = ''
+    cp -r ${finalAttrs.qui-web}/* web/dist
+  '';
+
+  ldflags = [
+    "-X github.com/autobrr/qui/internal/buildinfo.Version=${finalAttrs.version}"
+    "-X main.PolarOrgID="
+  ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgram = "${placeholder "out"}/bin/qui";
```

Comment:
`versionCheckProgram` is unneeded, because `meta.mainProgram` already sets this in a nicer manner